Core/include/Acts/Propagator/Propagator.ipp

```diff
@@ -256,6 +284,36 @@ auto Acts::Propagator<S, N>::makeResult(propagator_state_t state,
   moveStateToResult(state, result);
 
   if (makeCurvilinear) {
+    if constexpr (has_extension<decltype(state.stepping)>::value &&
+                  has_stepData<decltype(state.stepping)>::value) {
+      if (state.stepping.pathAccumulated <= 0.) {
+        // if no step was executed the path lenght derivates have not been
+        // computed but are needed to compute the curvilinear covariance. The
+        // derivates are given by k1 for a zero step width.
+        if (state.stepping.extension.validExtensionForStep(state, m_stepper,
+                                                           m_navigator)) {
+          // First Runge-Kutta point (at current position)
+          auto& sd = state.stepping.stepData;
+          auto pos = m_stepper.position(state.stepping);
+          auto fieldRes = m_stepper.getField(state.stepping, pos);
+          if (fieldRes.ok()) {
+            sd.B_first = *fieldRes;
+            if (state.stepping.extension.k1(state, m_stepper, m_navigator,
+                                            sd.k1, sd.B_first, sd.kQoP)) {
+              // dr/ds :
+              state.stepping.derivative.template head<3>() =
+                  state.stepping.pars.template segment<3>(eFreeDir0);
+              // d (dr/ds) / ds :
+              state.stepping.derivative.template segment<3>(4) = sd.k1;
+              // to set dt/ds :
+              state.stepping.extension.finalize(state, m_stepper, m_navigator,
+                                                state.stepping.pathAccumulated);
+            }
+          }
```

Comment:
do you think this can be done in the eigen stepper instead?

that way we can also avoid the template based checks if this applies or not

Reply:
You are right, It should go into the EigenStepper. Similar modifications would be needed in the AtlasStepper. It is less clear to me what is needed for the StraightLineStepper. I suppose the StraightLineStepper should produce the same as the EigenStepper in the limit B->0. However in this limit only the d (dr/ds) / ds are zero not the other components of the path length derivatives, and I do not see that being accounted for.